src/main/java/com/org/jenkins/custom/jenkins/distribution/service/services/UpdateCenterService.java

```diff
@@ -17,18 +20,40 @@
 
     private final static Logger LOGGER = Logger.getLogger(UpdateCenterService.class.getName());
     private static final String UPDATE_CENTER_JSON_URL = "https://updates.jenkins.io/current/update-center.actual.json";
-    
+    private int updateFlag = 0;
+    private String responseString;
+    private String updateCenterFilePath = "";
+
     public JSONObject downloadUpdateCenterJSON() throws Exception {
-        HttpGet get = new HttpGet(UPDATE_CENTER_JSON_URL);
-        LOGGER.info("Executing Request");
-        CloseableHttpClient httpClient = HttpClients.createDefault();
-        CloseableHttpResponse response = httpClient.execute(get);
-        String responseString = EntityUtils.toString(response.getEntity());
         /*
-        * Instead of writing to a file it would be better to return a json object as of now
-        * and later once we have the caching mechanism in place we could pull it in from there
+        * Check if updateFlag has been set if not then it is the first time the application
+        * is being run so we need to download the update-center
         */
+        System.out.println("Update flag is " + updateFlag);
+        System.out.println("Update Center Path " + updateCenterFilePath);
+
+        if(updateFlag == 0) {
+            File updateCenterFile = File.createTempFile("update-center", ".json");
+            updateCenterFilePath = updateCenterFile.getPath();
+            LOGGER.info("Creating a new file" + updateCenterFile.getPath());
+            HttpGet get = new HttpGet(UPDATE_CENTER_JSON_URL);
+            LOGGER.info("Executing Request");
```

Comment:
Might be useful to include that you're making a request to the update center url in the log message so it's easier to trace actions.

Reply:
Done :+1: